linera-execution/src/execution_state_actor.rs

```diff
@@ -445,10 +446,25 @@ where
                 callback.respond(index)
             }
 
-            ReadEvent { event_id, callback } => {
-                let event = self.context().extra().get_event(event_id.clone()).await?;
-                let event = event.ok_or(ExecutionError::EventsNotFound(vec![event_id]))?;
-                callback.respond(event);
+            ReadEvent {
+                event_id,
+                callback,
+                mut txn_tracker,
```

Comment:
TransactionTracker is quite big - is it OK to send it over a channel? Can't we verify it on the caller side?